src/fidesops/ops/service/strategy.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+import logging
+from abc import ABC
+from typing import Any, Dict, Generic, List, Optional, Type, TypeVar
+
+from pydantic import ValidationError
+
+from fidesops.ops.common_exceptions import NoSuchStrategyException
+from fidesops.ops.common_exceptions import ValidationError as FidesopsValidationError
+from fidesops.ops.schemas.saas.strategy_configuration import StrategyConfiguration
+
+logger = logging.getLogger(__name__)
+T = TypeVar("T", bound="Strategy")
+C = TypeVar("C", bound=StrategyConfiguration)
+
+
+def _find_strategy_subclass(
+    cls: Type[Strategy], strategy_name: str
+) -> Optional[Type[Strategy]]:
+    if hasattr(cls, "name") and cls.name == strategy_name:
+        return cls
+    for sub in cls.__subclasses__():
+        found = _find_strategy_subclass(sub, strategy_name)
+        if found is not None:
```

Comment:
I might do `if found` instead since I don't think the None comparison is needed specifically?